src/docs/cookbook-events/reacting-to-events/index.md

```diff
@@ -0,0 +1,402 @@
+    : { createdAtStart: startTime };
+  queryEvents(params).then(res => {
+    const events = res.data.data;
+    const fullPage = events.length === res.data.meta.perPage;
+    const delay = fullPage ? pollWait : pollIdleWait;
+    const lastEvent = events[events.length - 1];
+    const lastSequenceId = lastEvent
+      ? lastEvent.attributes.sequenceId
+      : sequenceId;
+
+    events.forEach(e => {
+      handleEvent(e);
+    });
+
+    setTimeout(() => {
+      pollLoop(lastSequenceId);
+    }, delay);
+  });
+};
+
+// Load state from local file, if any
+const lastSequenceId = loadLastEventSequenceId();
+
+// kick off the polling loop
+pollLoop(lastSequenceId);
+```
+
+## At-least-once and at-most-once event processing
+
+No system ever operates without failures and therefore applications
+should account for possible failure conditions during event processing.
+For instance, an API call to another system durung event handling may
+fail (sending notification, synchronizing state, executing other
+actions, etc), or the application itself may crash due to a bug or issue
+on the host system.
+
+An important aspect of this is what is often referred to as
+_at-least-once_ and _at-most-once_ processing in message processing
+systems.
+
+Consider the following scenarios:
+
+1. the application receives an event, records persistently its sequence
+   ID and then proceeds to handle the event
+2. the application receives an event, hadles it fully and then records
+   its sequence ID persistently
+
+Suppose that a failure occurrs during the event handling and the
+application crashes. In scenario 1, when the application is restarted,
+it will query for new events, skipping over the event that faild to be
+handled, as it's sequence ID is already stored. Depending on when
+exactly the failure occurred, the event may not have been fully handled
+and therefore will be left unprocessed (this is at-most-once
+processing). On the other hand, in scenario 2, when the application is
+restarted, it will retry handling the failed event (since the last
+recorded sequence ID is that of some previous event) and potentially may
+process the event a second time (that is, at-least-once processing).
+
+The preferable strategy for an application depends on the exact event
+handling logic. For example, at-least-once processing may be preferable
+in any of the following situations:
+
+- when the event processing is idempotent, meaning that processing and
+  event more than once does not produce undesired side-effects
+- when it is tolerrable that some events may be processed multiple times
+  (e.g. when a user may receive an occasional duplicate notification)
+
+On the other hand, at-most-once processing is be preferable if event the
+processing produces a side-effect that must not be repeated (such as
+making a money transfer) and can not be performed safely in an
+idempotent manner. In those cases, it may be desirable to handle the
```

Comment:
```suggestion
On the other hand, at-most-once processing is preferable if processing 
an event produces a side-effect that must not be repeated (such as
making a money transfer) and can not be performed safely in an
idempotent manner. In those cases, it may be desirable to handle the
```